### PR TITLE
Improve JSON objects conversion in services' classes

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/services/data/BatchService.java [hunk 1]
```diff
@@ -281,10 +281,12 @@ public void addAllObjectsToIndex() throws CustomResponseException, InterruptedEx
...
         BatchDTO batchDTO = new BatchDTO();
-        batchDTO.setTitle(getStringPropertyForDTO(jsonObject, "title"));
-        batchDTO.setType(getStringPropertyForDTO(jsonObject, "type"));
+        batchDTO.setId(getIdFromJSONObject(jsonObject));
+        JSONObject batchObject = getSource(jsonObject);
+        batchDTO.setTitle(getStringPropertyForDTO(batchObject, "title"));
+        batchDTO.setType(getStringPropertyForDTO(batchObject, "type"));
         if (!related) {
-            batchDTO = convertRelatedJSONObjects(jsonObject, batchDTO);
+            batchDTO = convertRelatedJSONObjects(batchObject, batchDTO);
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/data/BatchService.java]: batchObject sounds too much like it's a bean.
I would prefere batchJSONObject

File: Kitodo/src/main/java/org/kitodo/services/data/BatchService.java [hunk 2]
```diff
@@ -281,10 +281,12 @@ public void addAllObjectsToIndex() throws CustomResponseException, InterruptedEx
     @Override
     public BatchDTO convertJSONObjectToDTO(JSONObject jsonObject, boolean related) throws DataException {
         BatchDTO batchDTO = new BatchDTO();
-        batchDTO.setTitle(getStringPropertyForDTO(jsonObject, "title"));
-        batchDTO.setType(getStringPropertyForDTO(jsonObject, "type"));
+        batchDTO.setId(getIdFromJSONObject(jsonObject));
+        JSONObject batchObject = getSource(jsonObject);
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/services/data/BatchService.java]: why did you move the "getSource" out of the methods?

Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/data/BatchService.java]: I wanted to avoid this one level querying _source for every property. Probably it doesn't give much increase of the speed for one object, but for 1000 or 10000 it can already make a difference.

File: Kitodo/src/main/java/org/kitodo/services/data/BatchService.java (hunk 1 above)
Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/data/BatchService.java]: renamed
